src/test_utils/es/es_test_cluster.js

```diff
@@ -0,0 +1,87 @@
+import { resolve } from 'path';
+import { esTestServerUrlParts } from '../../../test/es_test_server_url_parts';
+
+import libesvm from 'libesvm';
+
+const ESVM_DIR = resolve(__dirname, '../../../esvm/test_utils/es_test_cluster');
+
+export class EsTestCluster {
+  _freshBranches = [];
+  _directory = resolve(ESVM_DIR);
+
+  use(options = {}) {
+    const {
+      name,
+      port = esTestServerUrlParts.port,
+      log = console.log,
+      branch = 'master',
+    } = options;
+
+    if (!name) {
+      throw new Error('esTestCluster.use() requires { name }');
+    }
+
+    // assigned in use.start(), reassigned in use.stop()
+    let cluster;
+
+    return {
+      start: async () => {
```

Comment:
I think `start` should throw an error if `cluster` is already defined, this way we don't get into a situation where calling start orphans a cluster object that was never shutdown.